### PR TITLE
added option in write.magpie for working with RasterBrick objects

### PR DESCRIPTION
**Pull request by flohump**
param x magclass or RasterBrick object. RasterBrick works only with file_types "nc", "asc", "grd" and "tif".

### CODE REVIEW

File: R/write.magpie.R [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #' data column. In the case that more than one year and data column is supplied
 #' several files are written with the structure filename_year_datacolumn.asc
 #'
-#' @param x MAgPIE-object
+#' @param x magclass or RasterBrick object. RasterBrick works only with file_types "nc", "asc", "grd" and "tif".
```
Comment (tscheypidi) [inline R/write.magpie.R:25]: It should be highlighted that magclass is the default here. Perhaps write something like:

"a magclass object. An exception is that formats written via the raster package (currently "nc", "asc", "grd" and "tif") also accept RasterBrick objects which have been previously created from a magclass object via as.RasterBrick)."

File: R/write.magpie.R [hunk 2]
```diff
@@ -99,7 +99,7 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
     mode <- as.character(777 - as.integer(as.character(umask)))
   }
   if (is.null(x)) x <- as.magpie(numeric(0))
-  if (is.magpie(x)) {
+  if (is.magpie(x) | class(x)  == "RasterBrick") {
```
Comment (tscheypidi) [inline R/write.magpie.R:102]: There needs to be a check somewhere that only the allowed file_types are being used when RasterBrick is chosen. Probably after line 107 which fills the file_type variable properly if it did not happen before. Or move the file_type step more to the front and then make a combined check here `(is.magpie(x) || (class(x) == "RasterBrick" && file_type %in% RasterFileTypes))`

File: R/write.magpie.R [hunk 3]
```diff
@@ -172,17 +172,28 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
...
+        varname <- getItems(x, dim = 3)
+        zunit <- ifelse(all(isYear(getYears(x))), "years", "")
+        x <- as.RasterBrick(x)
+      } else if (class(x) == "RasterBrick") {
+        tmp <- names(x)
+        tmp <- strsplit(tmp,"\\..")
+        years <- sort(unique(unlist(lapply(tmp,function(x) x[1]))))
+        varname <- sort(unique(unlist(lapply(tmp,function(x) x[2]))))
+        zunit <- ifelse(all(isYear(years)), "years", "")
+        years <- as.numeric(gsub("y","",years))
```
Comment (tscheypidi) [inline R/write.magpie.R:187]: is the `years` variable being used anywhere afterwards?

**Review by tscheypidi — CHANGES_REQUESTED** [review]

File: R/write.magpie.R (hunk 3 above)
Comment (flohump) [reply R/write.magpie.R]: No, it's not needed. Deleted.

File: R/write.magpie.R [hunk 4]
```diff
@@ -99,21 +101,24 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
...
   if (is.null(x)) x <- as.magpie(numeric(0))
-  if (is.magpie(x)) {
+  if (is.magpie(x) || class(x) == "RasterBrick") {
     years <- !(is.null(dimnames(x)[[2]]))
 
     # if file-type is not mentioned file-ending is used as file-type
     if (is.null(file_type)) {
       file_type <- tail(strsplit(file_name, "\\.")[[1]], 1) # nolint
     }
+    if (class(x) == "RasterBrick" && !file_type %in% c("nc", "asc", "grd", "tif")) {
```
Comment (tscheypidi) [inline R/write.magpie.R:111]:
```suggestion
    if (class(x) == "RasterBrick" && !(file_type %in% c("nc", "asc", "grd", "tif"))) {
```

**Review by tscheypidi — CHANGES_REQUESTED** [review]

**Review by tscheypidi — APPROVED** [review]

**Pull request merged**